pjmedia/src/pjmedia/codec.c

```diff
@@ -191,6 +249,8 @@ PJ_DEF(pj_status_t) pjmedia_codec_mgr_register_factory( pjmedia_codec_mgr *mgr,
     /* Re-sort codec based on priorities */
     sort_codecs(mgr);
 
+    sort_codec_id(mgr->codec_list, mgr->codec_list_cnt);
```

Comment:
just keep the array always sorted during insertion/deletion, so that we don't need to sort it over and over.